GoogleDataTransport/GDTCCTLibrary/GDTCORMetrics+GDTCCTSupport.m

```diff
@@ -0,0 +1,193 @@
+pb_size_t GDTCCTGetLogSourceMetricsCount(GDTCOREventMetricsCounter *droppedEventCounter) {
+  return (pb_size_t)droppedEventCounter.droppedEventCounterByMappingID.count;
+}
+
+/// Constructs and returns a ``gdt_client_metrics_TimeWindow`` proto from the given parameters.
+/// @param collectionStartDate The start of the time window.
+/// @param collectionEndDate The end of the time window.
+gdt_client_metrics_TimeWindow GDTCCTConstructTimeWindow(NSDate *collectionStartDate,
+                                                        NSDate *collectionEndDate) {
+  gdt_client_metrics_TimeWindow timeWindow = gdt_client_metrics_TimeWindow_init_zero;
+  // `- [NSDate timeIntervalSince1970]` returns a time interval in seconds so
+  // multiply by 1000 to convert to milliseconds.
+  timeWindow.start_ms = (int64_t)collectionStartDate.timeIntervalSince1970 * 1000;
+  timeWindow.end_ms = (int64_t)collectionEndDate.timeIntervalSince1970 * 1000;
+  return timeWindow;
+}
+
+/// Constructs and returns a ``gdt_client_metrics_GlobalMetrics`` proto from the given parameters.
+/// @param currentCacheSize The current cache size.
+/// @param maxCacheSize The max cache size.
+gdt_client_metrics_GlobalMetrics GDTCCTConstructGlobalMetrics(uint64_t currentCacheSize,
+                                                              uint64_t maxCacheSize) {
+  gdt_client_metrics_StorageMetrics storageMetrics = gdt_client_metrics_StorageMetrics_init_zero;
+  storageMetrics.current_cache_size_bytes = currentCacheSize;
+  storageMetrics.max_cache_size_bytes = maxCacheSize;
+
+  gdt_client_metrics_GlobalMetrics globalMetrics = gdt_client_metrics_GlobalMetrics_init_zero;
+  globalMetrics.storage_metrics = storageMetrics;
+
+  return globalMetrics;
+}
+
+/// Returns the corresponding ``gdt_client_metrics_LogEventDropped_Reason`` for the given
+/// ``GDTCOREventDropReason``.
+///
+/// To represent  ``GDTCOREventDropReason`` in a proto, the reason must be mapped to a
+/// ``gdt_client_metrics_LogEventDropped_Reason``.
+///
+/// @param reason The ``GDTCOREventDropReason`` to represent in a proto.
+gdt_client_metrics_LogEventDropped_Reason GDTCCTConvertEventDropReasonToProtoReason(
+    GDTCOREventDropReason reason) {
+  switch (reason) {
+    case GDTCOREventDropReasonUnknown:
+      return gdt_client_metrics_LogEventDropped_Reason_REASON_UNKNOWN;
+    case GDTCOREventDropReasonMessageTooOld:
+      return gdt_client_metrics_LogEventDropped_Reason_MESSAGE_TOO_OLD;
+    case GDTCOREventDropReasonStorageFull:
+      return gdt_client_metrics_LogEventDropped_Reason_CACHE_FULL;
+    case GDTCOREventDropReasonPayloadTooBig:
+      return gdt_client_metrics_LogEventDropped_Reason_PAYLOAD_TOO_BIG;
+    case GDTCOREventDropReasonMaxRetriesReached:
+      return gdt_client_metrics_LogEventDropped_Reason_MAX_RETRIES_REACHED;
+    case GDTCOREventDropReasonInvalidPayload:
+      return gdt_client_metrics_LogEventDropped_Reason_INVALID_PAYLOD;
```

Comment:
Is this the one misspelled on purpose?  If so, please comment.

Reply:
Yes, the proto field in g3 is misspelled. I will add a comment. Thanks!

Reply:
Done in 1af8d3b.